main.py

```diff
@@ -44,114 +54,132 @@ async def on_message(message: discord.Message):
-            else:
-                users.append(f"{mention_(user1, safe=True)} — {mention_(user2, safe=True)}")
-        else:
-            users.append(f"{mention_(user1)} — {mention_(user2)}")
+    s = ""
+    if len(pairings.paired) > 0:
+        s += "*1on1 pairs:*\n"
+        for pair in pairings.paired:
+            s += f"{mention_(guild, pair[0], asked=user_asked, always_ping=always_ping)} — {mention_(guild, pair[1], asked=user_asked, always_ping=always_ping)}\n"
+    elif len(pairings.unpaired) > 0:
+        s += "*waiting to be paired:*\n"
+        for user_id in pairings.unpaired:
+            s += f"{mention_(guild, user_id, asked=user_asked, always_ping=always_ping)}\n"
+    else:
+        s += "No pairings made yet"
 
     if ctx:
-        await ctx.respond(f"1on1 pairs: {', '.join(users)}")
+        await ctx.respond(s)
     elif channel:
-        await channel.send(f"1on1 pairs: {', '.join(users)}")
+        await channel.send(s)
+    else:
+        assert False # we need ctx or channel
 
 
-@bot.slash_command(guild_ids=[GUILD_ID])
-async def pair_1on1s(ctx):
-    # require admin
-    if not ctx.author.guild_permissions.administrator:
-        await ctx.respond("You must be an admin to run this command.")
-        return
-    pairs = await pair_for_1on1s(ctx.guild)
-    await display_pairs(ctx.guild, pairs, ctx=ctx)
-
 
 @bot.slash_command(guild_ids=[GUILD_ID])
-async def show_1on1_pairs(ctx):
-    channel = discord.utils.get(ctx.guild.channels, name="1on1-pairs")
-    last_message = channel.last_message
-    if last_message:
-        pairs_json = last_message.content
-        pairs = json.loads(pairs_json)
-        await display_pairs(ctx.guild, pairs, user_asked=ctx.author.id, ctx=ctx)
+async def show_1on1_pairs(ctx: discord.Interaction):
+    wps = await WeeklyPairings.load(ctx.guild)
+    if wps:
+        await display_pairs(ctx.guild, wps, user_asked=ctx.author.id, ctx=ctx)
     else:
         await ctx.respond("No pairs found")
 
+async def pair_weekly_users(guild: discord.Guild):
+    hist = await History.load_or_create_new(guild)
+    wps = await WeeklyPairings.load(guild)
+    if wps:
+        # also include people who were waiting to be paired last time
+        unpaired = set(wps.unpaired)
+    else:
+        unpaired = set()
+    opt_in = list(set([member.id for member in guild.members if discord.utils.get(member.roles, name="1on1")]) + unpaired)
+    pairs, unpaired = hist.pair_people(opt_in=opt_in)
+    wps = WeeklyPairings(unpaired=unpaired, paired=pairs)
+    c1 = wps.save(guild)
+    c2 = hist.save(guild)
+    await asyncio.gather(c1, c2)
+    return wps
+
+@bot.slash_command(guild_ids=[GUILD_ID])
+async def pairme(ctx: discord.Interaction):
+    hist = await History.load_or_create_new(ctx.guild)
+    pairs = await WeeklyPairings.load(ctx.guild)
+    person = ctx.author.id
+    if any(person in pair for pair in pairs.paired):
+        # TODO should we allow anyone to just sink all the pairs?
```

Comment:
Should we allow someone to spam `/pairme` and then 1on1 with so many people?

Reply:
yes i think

Reply:
Do you want me to remove this check then?